changelog.d/pr-8557-2

```diff
@@ -0,0 +1,16 @@
+synopsis: Add --semaphore option to ./Setup build interface
+packages: Cabal
+prs: #8557
+
+description: {
+
+When `./Setup build --semaphore <SEM>` is called, `ghc` will be called with the
+`-jsem` option. It is the responsibility of the caller of `./Setup build` to
+manage the semaphore according to the GHC Jobserver Protocol.
+
+This low level interface is intended to be called by a high-level tool such as
+`cabal-install` which can create and manage the semaphore appropriately.
```

Comment:
@ffaf1 this is what you've been seeing: cabal-install handles the uniqueness of the name and uses the more low-level interface of `./Setup build`